src/main/java/za/ac/cput/repository/IStudentRepository.java

```diff
@@ -8,11 +8,9 @@
 implementation
 */
 
+import org.springframework.data.jpa.repository.JpaRepository;
 import za.ac.cput.entity.Student;
-import java.util.Set;
 
-public interface IStudentRepository extends IRepository<Student, String>
+public interface IStudentRepository extends JpaRepository<Student, String>
```

Comment:
Check the type there. Your ID is of type long and yet here you specified a String type instead of a long.

You'll later run into a situation where you will have to convert between types which will become cumbersome.
Rather play safe and change String to Long as your ID's are of type Long.